plugins/modules/database.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2022, John McCall (@lowlydba)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+DOCUMENTATION = r'''
+---
+module: database
+short_description: Creates and configures a database.
+description:
+     - Adds a new database to an existing SQL Server instance.
+options:
+  database:
+    description:
+      - Name of the target database.
+    type: str
+    required: true
+  recovery_model:
+    description:
+      - Choose the recovery model for the database.
+    type: str
+    required: false
+    choices: ['Full', 'Simple', 'BulkLogged']
+  data_file_path:
+    description:
+      - Directory where the data files should be placed. Uses SQL Server's default if not supplied.
+        Only used if database is being created.
+    type: str
+    required: false
+  log_file_path:
+    description:
+      - Directory where the log files should be placed. Uses SQL Server's default if not supplied.
+        Only used if database is being created.
+    type: str
+    required: false
+  owner_name:
+    description:
+      - Database owner login
+    type: str
+    required: false
+  maxdop:
+    description:
+      - MAXDOP value for the database.
+    required: false
+    type: int
+  secondary_maxdop:
+    description:
+      - MAXDOP value for the database when it is a non-primary replica in an availability group.
+    required: false
+    type: int
+  compatibility:
+    description:
+      - Compatibility mode for the database. Follows the format of "Version90", "Version100", and so on.
+        String is validated by Set-DbaDbCompatibility.
+    required: false
+    type: str
+  rcsi:
+    description:
+      - Whether or not to enable Read Committed Snapshot Isolation.
+    required: false
+    type: bool
+  state:
+    description:
+      - Whether or not the database should be created, remain, or be dropped.
```

Comment:
kind of a nit but the description is a little confusing when compared to the `choices`. Maybe it should be like:
```suggestion
      - Whether or not the database should be C(present) or C(absent).
```
don't feel terribly strongly about it though.